demos/text-field.html

```diff
@@ -679,6 +731,56 @@ <h2>Full-Width Text Field and Textarea</h2>
           section.classList[target.checked ? 'add' : 'remove']('mdc-theme--dark');
         });
       });
+
+      demoReady(function() {
+
+        // Initializes all checkboxes to find closest "example" class (section), finds the elementToExecuteCallbackOn
+        // and calls a function with the event and each element found.
+        var findElementAndAddClickHandlerToCheckBox = function(checkboxClassName, elementToExecuteCallbackOn, handlerCallback) {
+          var checkboxes = Array.from(document.getElementsByClassName(checkboxClassName));
+
+          checkboxes.forEach(function (checkbox) {
+            checkbox.addEventListener('change', function (evt) {
+              var parentSection = (function (box) {
+                // walk up the dom to find the "example" class.
+                while (!box.classList.contains('example')) {
+                  if(box.parentElement === document.body) {
+                    return null;
+                  }
+                  box = box.parentElement;
+                }
+                return box;
+              })(checkbox);
+
+              parentSection.querySelectorAll(elementToExecuteCallbackOn)
```

Comment:
Does this even need to use qSA or is it only ever being used to operate on a single element?

If you do need qSA, wrap in `[].slice.call` to work in all browsers, e.g. `[].slice.call(parentSection.querySelectorAll(...)).forEach(...)`